client/src/store/slices/cartSlice/cartSlice.js

```diff
@@ -1,158 +1,248 @@
 /* eslint-disable no-underscore-dangle */
-/* eslint-disable no-return-assign */
-/* eslint-disable no-param-reassign */
-import { createSlice } from '@reduxjs/toolkit';
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
 import axiosConfig from '../../../axiosConfig';
-import getProduct from '../../../api/getProduct';
-import updateCart from '../../../api/updateCart';
+// import updateCart from '../../../api/updateCart';
+import writeToLocalStorage from '../../utils/writeToLocalStorage';
+
+const initialState = {
+  data: [],
+  isLoading: false,
+  error: null,
+};
+
+/**
+ * fetches online cart from DB
+ * @param {string} testLoad - a test payload since the createAsyncThunk functions doesnt work properly if not supplied with payload, to be inspected later.
+ * @returns an object which containts a products array
+ */
+export const fetchCart = createAsyncThunk('cart/fetchCart', async (testLoad, { rejectWithValue }) => {
+  try {
+    const { data } = await axiosConfig('/cart');
+    return data;
+  } catch (err) {
+    console.error(err.message);
+    return rejectWithValue('Something went wrong. Please reload the page or try again later.');
+  }
+});
+
+/**
+ * deletes online cart from DB
+ * @returns nothing
```

Comment:
I suggest using void instead of nothing or completely delete this return comment
```suggestion
 * @returns void
```